tasks.py

```diff
@@ -782,14 +785,89 @@ def show_automerges(ctx: Context):
         print("\tNo PRs set to automerge")
 
 
-MARKER_REQ_MAPPING: Final[Mapping[str, tuple[str, ...]]] = {
-    "athena": ("reqs/requirements-dev-athena.txt",),
-    "cloud": ("reqs/requirements-dev-cloud.txt",),
-    "pyarrow": ("reqs/requirements-dev-arrow.txt",),
-    "external_sqldialect": ("reqs/requirements-dev-sqlalchemy.txt",),
+class TestDependencies(NamedTuple):
+    requirement_files: tuple[str, ...]
+    setup_funcs: tuple[Callable[[Context], Result | None], ...] = tuple()
+    exta_pytest_args: tuple[  # TODO: remove this once remove the custom flagging system
+        str, ...
+    ] = tuple()
+
+
+MARKER_DEPENDENDENCY_MAP: Final[Mapping[str, TestDependencies]] = {
+    "athena": TestDependencies(("reqs/requirements-dev-athena.txt",)),
+    "cloud": TestDependencies(
+        ("reqs/requirements-dev-cloud.txt",), exta_pytest_args=("--cloud",)
+    ),
+    "external_sqldialect": TestDependencies(("reqs/requirements-dev-sqlalchemy.txt",)),
+    "pyarrow": TestDependencies(("reqs/requirements-dev-arrow.txt",)),
+    "postgresql": TestDependencies(
+        ("reqs/requirements-dev-postgresql.txt",),
+        setup_funcs=(
+            lambda ctx: ctx.run(
+                "docker-compose -f assets/docker/postgresql/docker-compose.yml up -d"
+            ),
```

Comment:
We should leverage the `service` task here so we don't up the service differently in these 2 places. 

```suggestion
            lambda ctx: service(ctx, names=["postgresql"]),
```

Reply:
I believe this was the other way around @billdirks. We kept `setup_funcs` and took out `services`

Reply:
Done. Thanks for catching this